ls1intum/Artemis
# `General`: Improve charts in course statistics and course overview

### Checklist
#### General



#### Client

### Motivation and Context


This pull request aims to address two issues with the current bar chart in the course statistics view and course overview. Firstly, the values for missed points are not rounded, leading to inconsistency. Secondly, the text within the chart on the course statistics page is not centered, affecting its visual appeal. By rounding the values and centering the text, this improvement enhances readability, providing a more visually pleasing experience for users.

### Description

In this pull request, I have made several improvements to enhance the doughnut chart in the course statistics view. Firstly, I centered the text by setting its width to match the width of the doughnut chart, improving its visual alignment. Secondly, I have ensured that all values in the chart are now rounded in accordance with the course settings, promoting consistency. Lastly, I addressed an issue on the course overview page where the center text was being cut off for larger point values. To resolve this, I decreased the size of the center text, ensuring that it remains fully visible even for larger point amounts.

### Steps for Testing

Prerequisites:
- 1 Student
- Multiple Courses with varying amounts of reached and missed points (you should be able to just verify that everything works by utilising existing courses on out test servers)

1. Log in as a user.
2. Verify that the doughnut charts on the course overview page have centered text that is not cut off.
3. Navigate to _your course_ -> statistics.
4. Verify that the text is centered in the doughnut chart .
5. Verify that you can hover over the individual parts of the chart and that the points are rounded.

### Review Progress

#### Code Review
#### Manual Tests

### Screenshots

#### Before:
[Course Overview] [Course Statistics]
<div style="display: flex">
<img width="360" alt="Bildschirmfoto 2023-06-15 um 09 25 03" src="https://github.com/ls1intum/Artemis/assets/47261058/3f324bb5-20c7-4f2c-9885-2ab08217c499">
<img width="360" alt="Bildschirmfoto 2023-06-15 um 09 03 31" src="https://github.com/ls1intum/Artemis/assets/47261058/e19dc771-68b5-4301-9373-72ce5a70da52">
</div>

#### After:
[Course Overview] [Course Statistics]
<div style="display: flex">
<img width="360" alt="Bildschirmfoto 2023-06-15 um 09 45 58" src="https://github.com/ls1intum/Artemis/assets/47261058/daf1f970-a538-4a09-879b-fd8684de0b63">
<img width="360" alt="Bildschirmfoto 2023-06-15 um 09 21 10" src="https://github.com/ls1intum/Artemis/assets/47261058/d0bce70e-2c1f-4cbc-bb48-474bdbac3648">
</div>
